bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/URLAudioStream.java

```diff
@@ -65,24 +66,20 @@ private InputStream createInputStream() throws AudioException {
         try {
             switch (extension) {
                 case M3U_EXTENSION:
-                    try (final InputStream isM3U = new URL(url).openStream()) {
-                        for (final String line : IOUtils.readLines(isM3U)) {
-                            if (!line.isEmpty() && !line.startsWith("#")) {
-                                url = line;
-                                break;
-                            }
+                    for (final String line : Files.readAllLines(new File(url).toPath())) {
```

Comment:
Converting urls to files can cause a lot of problems if the urls don't have the right schema so create the path from a URI instead.
```suggestion
                    for (final String line : Files.readAllLines(Paths.get(URI.create(url)))) {
```